script/reference-device/dhcp6_pd_daemon.py

```diff
@@ -107,8 +107,17 @@ def connect_to_signal():
 
 
 def check_and_reconnect(dbus_obj):
+    global bus
     if dbus_obj is None:
         connect_to_signal()
+    bus.watch_name_owner(bus_name='io.openthread.BorderRouter.wpan0',
```

Comment:
For this [watch_name_owner API](https://dbus.freedesktop.org/doc/dbus-python/dbus.bus.html?highlight=watch_name_owner#dbus.bus.BusConnection.watch_name_owner), what's the behavior of calling it multiple times? I failed to find detailed description regarding it.

Reply:
The [dbus watch implementation](https://dbus.freedesktop.org/doc/api/html/structDBusWatch.html) shows dbus lib keeps internal lists of watch entries. When a new watch is registered for the same name and the same callback, it simply updates the existing entry in the table.

Reply:
The [source code](https://github.com/TelepathyIM/telepathy-glib/blob/72412c944b771f3214ddc40fa9dea82cea3a5651/telepathy-glib/dbus-daemon.c#L519C7-L519C26) shows more details